### PR TITLE
DoesRangeInclude function

### PR DESCRIPTION
**Pull request by rmetzman**
# Description of the PR

- Add functionality for DoesRangeInclude: given a range of versions and an array of versions, will return true if any of the versions are contained within the range and false if not
- Add testing for DoesRangeInclude

# PR Checklist

- [ x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x ] All new changes are covered by tests
- [ x] If GraphQL schema is changed, `make generate` has been run
- [ x] If `collectsub` protobuf has been changed, `make proto` has been run
- [ x] All CI checks are passing (tests and formatting)
- [ x] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/misc/depversion/depversion_test.go [hunk 1]
```diff
@@ -442,3 +442,41 @@ func Test_WhichVersionMatches(t *testing.T) {
...
+		{
+			versionRange: ">=1.0,<=2.0",
+			versions:     []string{"1.0", "2.0", "3.0"},
+			expect:       true,
+		},
+		{
+			versionRange: ">=1.0,<=2.0",
+			versions:     []string{"3.0", "2.1"},
+			expect:       false,
+		},
```
Comment (lumjjb) [inline pkg/misc/depversion/depversion_test.go:466]: will be good to have several test cases that check for errors as well

**Review by lumjjb — APPROVED** [review]
couple nits else LGTM!

File: pkg/misc/depversion/depversion_test.go (hunk 1 above)
Comment (rmetzman) [reply pkg/misc/depversion/depversion_test.go:466]: just added a couple tests for errors, wasn't sure if its fine to include it in a seperate test

File: pkg/misc/depversion/depversion.go [hunk 2]
```diff
@@ -116,9 +116,23 @@ func ParseVersionValue(s string) VersionValue {
...
+	versionMap, err := WhichVersionMatches(versions, versionRange)
+	for k, v := range versionMap {
+		fmt.Println(k, "value is", v)
+	}
+
+	if err != nil {
+		return false, fmt.Errorf("error for DoesRangeInclude %v", err)
+	}
+
+	for _, isInRange := range versionMap {
```
Comment (pxp928) [inline pkg/misc/depversion/depversion.go:129]: the value will always be true. We could just check the length and if > 0 and return true

**Review by pxp928 — APPROVED** [review]
LGTM! Couple of comments

File: pkg/misc/depversion/depversion.go [hunk 3]
```diff
@@ -116,9 +116,23 @@ func ParseVersionValue(s string) VersionValue {
 	return vv
 }
 
-// TODO: implement for more efficient traversal later
 func DoesRangeInclude(versions []string, versionRange string) (bool, error) {
-	return false, fmt.Errorf("unimplemented")
+	versionMap, err := WhichVersionMatches(versions, versionRange)
+	for k, v := range versionMap {
+		fmt.Println(k, "value is", v)
```
[the remark this answers is not in the document]
Comment (rmetzman) [reply pkg/misc/depversion/depversion.go]: Done.

File: pkg/misc/depversion/depversion.go (hunk 2 above)
Comment (rmetzman) [reply pkg/misc/depversion/depversion.go]: Done.

**Pull request merged**